src/test/java/net/snowflake/client/jdbc/ResultSetLatestIT.java

```diff
@@ -924,4 +925,46 @@ public void testGranularTimeFunctionsInUTC() throws SQLException {
       connection.close();
     }
   }
+
+  @Test
+  public void testLargeStringRetrieval() throws SQLException {
+    Connection con = getConnection();
+    Statement statement = con.createStatement();
+    statement.execute("create or replace table maxJsonStringLength_table (c1 string(16777216))");
+    statement.execute("insert into maxJsonStringLength_table select randstr(16777216, random())");
+    try {
+      assertNull(System.getProperty("net.snowflake.jdbc.objectMapper.maxJsonStringLength"));
+      ResultSet rs = statement.executeQuery("select * from maxJsonStringLength_table");
+      assertNotNull(rs);
+    } catch (Exception e) {
+      fail("executeQuery should not fail");
+    } finally {
+      statement.execute("drop table maxJsonStringLength_table");
+      statement.close();
+      con.close();
+    }
+  }
+
+  @Test
+  public void testInvalidMaxJsonStringLength() throws SQLException {
+    Connection con = null;
+    Statement statement = null;
+    System.setProperty("net.snowflake.jdbc.objectMapper.maxJsonStringLength", "abc");
+    try {
+      // calling getObjectMapper() should log the exception but not throw
+      ObjectMapperFactory.getObjectMapper();
+      con = getConnection();
+      statement = con.createStatement();
+      statement.execute("create or replace table maxJsonStringLength_table (c1 string(16777216))");
```

Comment:
let's add a test showing that we can set the length in configuration to the lower value (e.g. the current jackson default) and check that it fails

Reply:
Added a test case in `ObjectMapperTest` and will be tested by `testObjectMapperWithLargeJsonString`